x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Space.java

```diff
@@ -0,0 +1,126 @@
+import org.elasticsearch.compute.operator.BreakingBytesRefBuilder;
+import org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator;
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.expression.Literal;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.expression.function.Example;
+import org.elasticsearch.xpack.esql.expression.function.FunctionInfo;
+import org.elasticsearch.xpack.esql.expression.function.Param;
+import org.elasticsearch.xpack.esql.expression.function.scalar.UnaryScalarFunction;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamInput;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.elasticsearch.common.unit.ByteSizeUnit.MB;
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.ParamOrdinal.DEFAULT;
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.isType;
+import static org.elasticsearch.xpack.esql.core.type.DataType.KEYWORD;
+
+public class Space extends UnaryScalarFunction {
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(Expression.class, "Space", Space::new);
+
+    static final long MAX_LENGTH = MB.toBytes(1);
+
+    @FunctionInfo(
+        returnType = "keyword",
+        description = "Returns a string of `number` spaces.",
+        examples = @Example(file = "string", tag = "space")
+    )
+    public Space(
+        Source source,
+        @Param(name = "number", type = { "integer" }, description = "Number of spaces in result.") Expression number
+    ) {
+        super(source, number);
+    }
+
+    private Space(StreamInput in) throws IOException {
+        this(Source.readFrom((PlanStreamInput) in), in.readNamedWriteable(Expression.class));
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        source().writeTo(out);
+        out.writeNamedWriteable(field);
+    }
+
+    @Override
+    public String getWriteableName() {
+        return ENTRY.name;
+    }
+
+    @Override
+    public DataType dataType() {
+        return KEYWORD;
+    }
+
+    @Override
+    protected TypeResolution resolveType() {
+        if (childrenResolved() == false) {
+            return new TypeResolution("Unresolved children");
+        }
+
+        return isType(field, dt -> dt == DataType.INTEGER, sourceText(), DEFAULT, "integer");
+    }
+
+    @Evaluator(warnExceptions = { IllegalArgumentException.class })
+    static BytesRef process(@Fixed(includeInToString = false, build = true) BreakingBytesRefBuilder scratch, int number) {
+        checkNumber(number);
+        scratch.grow(number);
+        scratch.clear();
+        for (int i = 0; i < number; ++i) {
+            scratch.append((byte) ' ');
+        }
```

Comment:
nit: I believe you can bulk-fill with:
```java
scratch.setLength(number);
Arrays.fill(scratch.bytes(), 0, number, ' ');
```
It feels more semantic to me, and potentially optimized

Reply:
Done.